build-tests/api-extractor-scenarios/etc/test-outputs/typeOf/api-extractor-scenarios.api.json

```diff
@@ -81,7 +82,8 @@
             },
             {
               "kind": "Reference",
-              "text": "ForgottenExport"
+              "text": "ForgottenExport",
+              "canonicalReference": "!ForgottenExport:class"
```

Comment:
`ForgottenExport` is linked as a global (ambient) declaration.  But it is effectively an unreachable declaration.  It would be more correct to omit the `canonicalReference` entirely.  (It could also be modeled using JSDoc's `~` "inner member" notation.  But currently API Extractor has the philosophy that if something is important enough to be documented, it should be importable.)

Reply:
That philosophy may not match with TypeScript output as we may sometimes manufacture non-exported types in declaration files as a result of inference.

Reply:
I've improved the declaration reference generation for local (non-exported) types.

Reply:
> we may sometimes manufacture non-exported types in declaration files as a result of inference.

Are you referring to [this sort of thing](https://github.com/microsoft/web-build-tools/blob/16c6a7e36f5d67c4aae8dda17b7052e6ddb3ce4e/apps/api-extractor-model/src/model/ApiConstructSignature.ts#L66), where `extends ApiTypeParameterListMixin(ApiParameterListMixin(ApiReleaseTagMixin(
  ApiReturnTypeMixin(ApiDeclaredItem))))` gets separated into `ApiConstructor_base`?

**lib/model/ApiConstructor.d.ts**
```ts
declare const ApiConstructor_base: typeof ApiDeclaredItem & (new (...args: any[]) => ApiReleaseTagMixin) & (new (...args: any[]) => ApiParameterListMixin);

/**
 * Represents a TypeScript class constructor declaration that belongs to an `ApiClass`.
 * @public
 */
export declare class ApiConstructor extends ApiConstructor_base {
    static getContainerKey(overloadIndex: number): string;
    constructor(options: IApiConstructorOptions);
    /** @override */
    readonly kind: ApiItemKind;
    /** @override */
    readonly containerKey: string;
    /** @beta @override */
    buildCanonicalReference(): DeclarationReference;
}
```

Reply:
That is one case, though I think there are others as well.